app/presenters/tree_node/menu/item.rb

```diff
@@ -0,0 +1,36 @@
+module TreeNode
+  module Menu
+    class Item < TreeNode::Menu::Node
+      set_attribute(:key)      { "#{@options[:node_id_prefix]}__#{@object.feature}" }
+      set_attribute(:title)    { _(details[:name]) }
+      set_attribute(:tooltip)  { _(details[:description]) || _(details[:name]) }
+      set_attribute(:selected) { self_selected? || select_by_children_selected }
+
+      def children
```

Comment:
I agree it's very inefficient, but the alternative is to build the entire tree in a hash in the TreeBuilder so that everything can be counted, and I thought you didn't want that.